service/src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -633,16 +649,20 @@ private void validateUniqueAiNotebookInstance(
       ControlledAiNotebookInstanceResource notebookResource) {
     // Workspace ID is a proxy for project ID, which works because there is a permanent, 1:1
     // correspondence between workspaces and GCP projects.
+    // Note: since resource_type column is not guaranteed to be filled in, we scan by
```

Comment:
Why do we have this redundancy in our schema? If we can compute column C based on columns A and B, then why store C in the first place?

Reply:
My goal is to only store C and remove A and B. Right now, A and B can represent invalid states. I would rather code that out of existence. I wrote this up in more detail in the implementation notes [here](https://docs.google.com/document/d/1NE0NO0sojGJMU-AK7dNTboEExX6bMRUVOLzonZ8Nujw/edit#heading=h.ji383l81blkk)